x/commitment/keeper/msg_server_cancel_vest.go

```diff
@@ -29,19 +29,17 @@ func (k msgServer) CancelVest(goCtx context.Context, msg *types.MsgCancelVest) (
 	for i := 0; i < len(commitments.VestingTokens) && !remainingToCancel.IsZero(); i++ {
 		vesting := commitments.VestingTokens[i]
 
-		if vesting.Denom == msg.Denom {
-			cancelAmount := sdk.MinInt(remainingToCancel, vesting.UnvestedAmount)
+		cancelAmount := sdk.MinInt(remainingToCancel, vesting.UnvestedAmount)
 
-			vesting.TotalAmount = vesting.TotalAmount.Sub(cancelAmount)
-			vesting.UnvestedAmount = vesting.UnvestedAmount.Sub(cancelAmount)
+		vesting.TotalAmount = vesting.TotalAmount.Sub(cancelAmount)
+		vesting.UnvestedAmount = vesting.UnvestedAmount.Sub(cancelAmount)
 
-			if vesting.TotalAmount.IsZero() {
-				commitments.VestingTokens = append(commitments.VestingTokens[:i], commitments.VestingTokens[i+1:]...)
-				i--
-			}
-
-			remainingToCancel = remainingToCancel.Sub(cancelAmount)
+		if vesting.TotalAmount.IsZero() {
+			commitments.VestingTokens = append(commitments.VestingTokens[:i], commitments.VestingTokens[i+1:]...)
+			i--
```

Comment:
I am not sure what this does, wouldn't it add infinite loop potential?

Reply:
@jelysn Each trader will have a list of vesting schedule. So its an array. The code piece is to remove the vesting amount 0 item.

Reply:
@kenta-elys can you do instead something like that:

```
newVestingTokens := make([]*types.VestingToken, 0, len(commitments.VestingTokens))

for _, vesting := range commitments.VestingTokens {
    cancelAmount := sdk.MinInt(remainingToCancel, vesting.UnvestedAmount)
    vesting.TotalAmount = vesting.TotalAmount.Sub(cancelAmount)
    vesting.UnvestedAmount = vesting.UnvestedAmount.Sub(cancelAmount)

    if !vesting.TotalAmount.IsZero() {
        newVestingTokens = append(newVestingTokens, vesting)
    }

    remainingToCancel = remainingToCancel.Sub(cancelAmount)
    if remainingToCancel.IsZero() {
        break
    }
}

commitments.VestingTokens = newVestingTokens
```

that way we avoid manipulating loop index and side effects of modifying slice in place.